src/OAuth1.php

```diff
@@ -335,17 +335,28 @@ public function signRequest($request, $token = null)
      */
     protected function composeSignatureBaseString($method, $url, array $params)
     {
+        $url = isset($url) ? $url : '';
+
         if (strpos($url, '?') !== false) {
             list($url, $queryString) = explode('?', $url, 2);
             parse_str($queryString, $urlParams);
             $params = array_merge($urlParams, $params);
         }
         unset($params['oauth_signature']);
         uksort($params, 'strcmp'); // Parameters are sorted by name, using lexicographical byte value ordering. Ref: Spec: 9.1.1
+
+        $parts0 = strtoupper($method);
+        $parts0 = isset($parts0) ? $parts0 : '';
```

Comment:
Won't it always be set cause it's defined above?

Reply:
fixed